pkg/chain/ethereum/tbtc.go

```diff
@@ -1176,3 +1322,44 @@ func parseWalletAction(value uint8) (tbtc.WalletAction, error) {
 		return 0, fmt.Errorf("unexpected wallet action value: [%v]", value)
 	}
 }
+
+func (tc *TbtcChain) ValidateDepositSweepProposal(
+	proposal *tbtc.DepositSweepProposal,
+	depositsExtraInfo []struct {
+		*tbtc.Deposit
+		FundingTx *bitcoin.Transaction
+	},
+) error {
+	dei := make([]tbtcabi.WalletCoordinatorDepositExtraInfo, len(depositsExtraInfo))
+	for i, depositExtraInfo := range depositsExtraInfo {
+		fundingTx := tbtcabi.BitcoinTxInfo2{
+			Version:      depositExtraInfo.FundingTx.SerializeVersion(),
+			InputVector:  depositExtraInfo.FundingTx.SerializeInputs(),
+			OutputVector: depositExtraInfo.FundingTx.SerializeOutputs(),
+			Locktime:     depositExtraInfo.FundingTx.SerializeLocktime(),
+		}
+
+		dei[i] = tbtcabi.WalletCoordinatorDepositExtraInfo{
+			FundingTx:        fundingTx,
+			BlindingFactor:   depositExtraInfo.Deposit.BlindingFactor,
+			WalletPubKeyHash: depositExtraInfo.Deposit.WalletPublicKeyHash,
+			RefundPubKeyHash: depositExtraInfo.Deposit.RefundPublicKeyHash,
+			RefundLocktime:   depositExtraInfo.Deposit.RefundLocktime,
+		}
+	}
+
+	valid, err := tc.walletCoordinator.ValidateDepositSweepProposal(
+		convertDepositSweepProposalToAbiType(proposal),
+		dei,
+	)
+	if err != nil {
+		return fmt.Errorf("validation failed: [%v]", err)
+	}
+
+	// Should never happen but just in case.
```

Comment:
Let's explain for future-us why this should never happen (`ValidateDepositSweepProposal` reverts or returns true).

Reply:
Sure, see e68ed993